yt/data_objects/tests/test_fluxes.py

```diff
@@ -115,10 +109,18 @@ def test_correct_output_unit():
     sur = ds.surface(sp1,"HI_Density", .5*Nmax)
     sur['x'][0]
 
-@requires_file(ISOGAL)
+def test_correct_output_unit_fake_ds():
+    # implementing test_correct_output_unit() with fake dataset
+    ds = fake_random_ds(64, nprocs=4, particles=16**3)
+    x = y = z = .5
+    sp1 = ds.sphere((x, y, z), (300, 'kpc'))
+    Nmax = sp1.max('density')
+    sur = ds.surface(sp1, "density", .5*Nmax)
+    sur['x'][0]
+
 def test_radius_surface():
     # see #1407
-    ds = load(ISOGAL)
+    ds = fake_random_ds(64, nprocs=4, particles=16**3)
```

Comment:
Can you set the `length_unit` to a non-default value, something like `length_unit=10`? That will make sure the units of `code_length` are different from the default unit of the radius field (`'cm'`), which is what this test is trying to look for.

Reply:
done